plugins/com.google.cloud.tools.eclipse.appengine.newproject.maven/src/com/google/cloud/tools/eclipse/appengine/newproject/maven/CreateMavenBasedAppEngineStandardProject.java

```diff
@@ -64,8 +69,15 @@ protected void execute(IProgressMonitor monitor)
         ? groupId : this.packageName;
     List<IProject> archetypeProjects = projectConfigurationManager.createArchetypeProjects(location,
         getArchetypeDescriptor(), groupId, artifactId, version, packageName, properties,
-        importConfiguration, progress.newChild(60));
+        importConfiguration, progress.newChild(40));
 
+    for (IProject project : archetypeProjects) {
+      IFacetedProject facetedProject = ProjectFacetsManager.create(
```

Comment:
Normally you'd need to rebalance the SubMonitor with `setWorkRemaining()` as the number of `archetypeProjects` could be arbitrary. Here we are assuming that `createArchetypeProjects()` returns a single project, and you're using the `for()` out of convenience.